source/geometries/BlackBox.cc

```diff
@@ -0,0 +1,210 @@
+// ----------------------------------------------------------------------------
+//  nexus | BlackBox.cc
+//  DICE in a black box.
+// ----------------------------------------------------------------------------
+
+#include "BlackBox.h"
+#include "BlackBoxSiPMBoard.h"
+#include "FactoryBase.h"
+
+#include "MaterialsList.h"
+#include "IonizationSD.h"
+#include "OpticalMaterialProperties.h"
+#include "Visibilities.h"
+
+#include <G4Box.hh>
+#include <G4Tubs.hh>
+#include <G4GenericMessenger.hh>
+#include <G4Orb.hh>
+#include <G4NistManager.hh>
+#include <G4LogicalVolume.hh>
+#include <G4PVPlacement.hh>
+#include <G4Material.hh>
+#include <G4ThreeVector.hh>
+#include <G4VisAttributes.hh>
+#include <G4Colour.hh>
+#include <G4SDManager.hh>
+#include <G4UserLimits.hh>
+#include <Randomize.hh>
+#include <G4LogicalSkinSurface.hh>
+#include <G4OpticalSurface.hh>
+#include <G4RotationMatrix.hh>
+#include <G4SystemOfUnits.hh>
+
+namespace nexus {
+
+  REGISTER_CLASS(BlackBox, GeometryBase)
+
+  BlackBox::BlackBox():
+    GeometryBase(),
+    world_z_ (3. * m),
+    world_xy_ (2. *m),
+    box_z_ (2. * m),
+    box_xy_ (1. *m),
+
+    visibility_(0)
+  {
+    msg_ = new G4GenericMessenger(this, "/Geometry/BlackBox/",
+				  "Control commands of BlackBox.");
+    msg_->DeclareProperty("visibility", visibility_, "Giant detectors visibility");
+
+    G4GenericMessenger::Command&  specific_vertex_X_cmd =
```

Comment:
Please, change these 3 parameters to one, using `DeclarePropertyWithUnit` (you can look at the `NextNew` class as an example).

Reply:
Done!